neural_modelling/src/neuron/plasticity/common/maths.h

```diff
@@ -11,14 +11,22 @@
 #define MAX(X,Y) ((X) > (Y) ? (X) : (Y))
 
 //---------------------------------------
-// Function declarations
-//---------------------------------------
-address_t maths_copy_int16_lut(address_t start_address, uint32_t num_entries,
-                               int16_t *lut);
-
-//---------------------------------------
 // Plasticity maths function inline implementation
 //---------------------------------------
+static inline address_t maths_copy_int16_lut(
+       address_t start_address, uint32_t num_entries, int16_t *lut) {
+
+    // Pad to number of words
+    const uint32_t num_words = (num_entries / 2)
+                               + (((num_entries & 1) != 0) ? 1 : 0);
+
+    // Copy entries to LUT
+    spin1_memcpy(lut, start_address, sizeof(int16_t) * num_entries);
+
+    // Return address after words
+    return start_address + num_words;
+}
+
```

Comment:
Yes, I did see this as potentially an issue.  My only real reason for this was to make the Makefile a bit nicer, but if it is going to be contentious, I don’t mind moving it back again, although the code bloat from those few lines is probably not going to be much!

One thought is to move this in to a library somewhere instead (e.g. spinn_common).

From: neworderofjamie [mailto:notifications@github.com]
Sent: 02 October 2015 14:09
To: SpiNNakerManchester/sPyNNaker <sPyNNaker@noreply.github.com>
Cc: Andrew Rowley <Andrew.Rowley@manchester.ac.uk>
Subject: Re: [sPyNNaker] Split input from model (#145)


In neural_modelling/src/neuron/plasticity/common/maths.h<https://github.com/SpiNNakerManchester/sPyNNaker/pull/145#discussion_r41018902>:

>  // Plasticity maths function inline implementation

>  //---------------------------------------

> +static inline address_t maths_copy_int16_lut(

> +       address_t start_address, uint32_t num_entries, int16_t *lut) {

> +

> +    // Pad to number of words

> +    const uint32_t num_words = (num_entries / 2)

> +                               + (((num_entries & 1) != 0) ? 1 : 0);

> +

> +    // Copy entries to LUT

> +    spin1_memcpy(lut, start_address, sizeof(int16_t) * num_entries);

> +

> +    // Return address after words

> +    return start_address + num_words;

> +}

> +

What was the purpose of this? It wasn't inline because it's not performance critical and is called in multiple places so will potentially bloat the code size!

—
Reply to this email directly or view it on GitHub<https://github.com/SpiNNakerManchester/sPyNNaker/pull/145/files#r41018902>.